jax/_src/numpy/fft.py

```diff
@@ -55,7 +58,18 @@ def _fft_core(func_name, fft_type, a, s, axes, norm):
     axes = tuple(range(a.ndim - len(axes), a.ndim))
     a = jnp.moveaxis(a, orig_axes, axes)
 
-  if s is None:
+  if s is not None:
+    a = jnp.asarray(a)
+    in_s = list(a.shape)
+    for axis, x in zip(axes, s):
+      in_s[axis] = x
```

Comment:
Use `safe_zip` to raise an error if `axes` and `s` have different sizes. I *think* this is handled by the above logic, but it's not obvious from a quick read so `safe_zip` would be safer :grin:

Reply:
It will throw "Shape and axes have different lengths.", but no problem I can add this :grinning: